src/test/java/it/gov/pagopa/payment/controller/TransactionControllerTest.java

```diff
@@ -0,0 +1,40 @@
+package it.gov.pagopa.payment.controller;
+
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.service.TransactionService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(TransactionControllerImpl.class)
+class TransactionControllerTest {
+
+    @MockBean
+    private TransactionService transactionService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String TRANSACTION_ID = "TRANSACTIONID1";
+
+    private static final String USER_ID = "USERID1";
+
+    @Test
+    void getTransaction() throws Exception {
+        when(transactionService.getTransaction(any(String.class), any(String.class)))
+                .thenReturn(new TransactionInProgress());
```

Comment:
You are checking that the service has been invoked, but you are not checking if its returned value is used as supposed:

1. return a mocked instance using TransactionInProgressFaker
2. after the mockMvc call, use its returned value to obtain the content as String
3. add an instance variable of type ObjectMapper (make it final) initiatiated calling it.gov.pagopa.payment.configuration.JsonConfig#objectMapper
4. add an instance variable of it.gov.pagopa.payment.dto.mapper.TransactionInProgress2TransactionResponseMapper (make it final)
5. Assert that the objectMapper.readValue of the body is equal to the mapper.apply(expectedTransactionInProgress)